Feat: React 18 support?

**Is your feature request related to a problem? Please describe.**
When using `react@18.0.0` and running `npm install`, an error gets thrown.

**Describe the solution you'd like**
Just update `peerDependencies` and test out dropdown

**Describe alternatives you've considered**
Running `npm config set legacy-peer-deps true` solves the problem.

**Additional context**
Logs:
```
npm ERR! Found: react@18.0.0
npm ERR! node_modules/react
npm ERR!   react@"^18.0.0" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer react@"^16.0.0 || ^17" from react-dropdown-tree-select@2.7.1
npm ERR! node_modules/react-dropdown-tree-select
npm ERR!   dev react-dropdown-tree-select@"2.7.1" from the root project
npm ERR! 
npm ERR! Fix the upstream dependency conflict, or retry
npm ERR! this command with --force, or --legacy-peer-deps
npm ERR! to accept an incorrect (and potentially broken) dependency resolution.
```

Using Node 16, npm 8.2.0

--- a/package.json
+++ b/package.json
@@ -100,7 +100,7 @@
     "prettier-stylelint": "^0.4.2",
     "prop-types": "^15.6.0",
     "raw-loader": "^0.5.1",
-    "react": "^16.0.0",
+    "react": "^16.3.0",
     "react-dom": "^16.0.0",
     "react-story": "^0.0.10",
     "rimraf": "^2.6.1",
@@ -116,7 +116,7 @@
     "why-did-you-update": "^0.1.1"
   },
   "peerDependencies": {
-    "react": "^16.0.0 || ^17"
+    "react": "^16.3.0 || ^17 || ^18"
   },
   "ava": {
     "files": [

--- a/src/index.js
+++ b/src/index.js
@@ -106,7 +106,7 @@ class DropdownTreeSelect extends Component {
     }
   }
 
-  componentWillMount() {
+  UNSAFE_componentWillMount() {
     this.initNewProps(this.props)
   }
 
@@ -114,7 +114,7 @@ class DropdownTreeSelect extends Component {
     document.removeEventListener('click', this.handleOutsideClick, false)
   }
 
-  componentWillReceiveProps(nextProps) {
+  UNSAFE_componentWillReceiveProps(nextProps) {
     this.initNewProps(nextProps)
   }
 

--- a/src/tree/index.js
+++ b/src/tree/index.js
@@ -38,7 +38,7 @@ class Tree extends Component {
     }
   }
 
-  componentWillReceiveProps = nextProps => {
+  UNSAFE_componentWillReceiveProps = nextProps => {
     const { activeDescendant } = nextProps
     const hasSameActiveDescendant = activeDescendant === this.props.activeDescendant
     this.computeInstanceProps(nextProps, !hasSameActiveDescendant)

